st2actions/st2actions/notifier/notifier.py

```diff
@@ -167,7 +167,15 @@ def _post_notify_subsection_triggers(self, liveaction=None, execution=None,
             payload['execution_id'] = execution_id
             payload['status'] = liveaction.status
             payload['start_timestamp'] = isotime.format(liveaction.start_timestamp)
-            payload['end_timestamp'] = isotime.format(liveaction.end_timestamp)
+
+            try:
+                payload['end_timestamp'] = isotime.format(liveaction.end_timestamp)
+            except AttributeError:
+                # This can be raised if liveaction.end_timestamp is None, which is caused
+                # when policy cancels a request due to concurrency
+                # In this case, use the start_timestamp instead
+                payload['end_timestamp'] = isotime.format(liveaction.start_timestamp)
```

Comment:
Good idea - updated in https://github.com/StackStorm/st2/pull/2985/commits/5f58b0414c0906ffb102350b09a115098d5abcd4